biorxiv/polka_subset_experiment/03_rerun_analyses_with_added_subset.py

```diff
@@ -0,0 +1,602 @@
+
+# Graph here?
+plt.figure(figsize=(8, 5))
+g = sns.violinplot(
+    x="version_count",
+    y="days_to_published",
+    data=biorxiv_published_distances,
+    cut=0,
+    scale="width",
+    palette="YlGnBu",
+)
+_ = g.set_ylabel("Time Elapsed Until Preprint is Published (Days)")
+_ = g.set_xlabel("# of Preprint Versions")
+_ = g.plot(x_line - 1, y_line, "--k")
+_ = g.scatter(
+    polka_published_preprint_df["version_count"] - 1,
+    polka_published_preprint_df["days_to_published"],
+    c="red",
+    s=12,
+)
+_ = g.annotate(f"Y={results_2.slope:.2f}*X+{results_2.intercept:.2f}", (7, 1470))
+_ = g.set_xlim(-0.5, 11.5)
+_ = g.set_ylim(0, g.get_ylim()[1])
+plt.savefig("output/figures/version_count_vs_publication_time_violin.svg", dpi=500)
+plt.savefig("output/figures/version_count_vs_publication_time_violin.png", dpi=500)
+
+# ### Document embedding pair
+
+polka_published_df = polka_published_df.set_index("document")
+polka_published_df.head()
+
+polka_preprints_df = polka_preprints_df.set_index("biorxiv_base").drop(
+    ["document", "version_count"], axis=1
+)
+polka_preprints_df.head()
+
+dist = np.diag(
+    cdist(
+        polka_preprints_df.loc[polka_published_preprint_df["biorxiv_base"]],
+        polka_published_df.loc[polka_published_preprint_df["PMCID"]],
+    )
+)
+print(dist)
+
+polka_published_preprint_df = polka_published_preprint_df.assign(doc_distances=dist)
+polka_published_preprint_df.head()
+
+# +
+# Get smoothed linear regression line
+x = biorxiv_published_distances.doc_distances.values.tolist()
+
+y = biorxiv_published_distances.time_to_published.apply(
+    lambda x: x / timedelta(days=1)
+).tolist()
+
+xseq_2 = np.linspace(np.min(x), np.max(x), 80)
+
+results_2 = linregress(x, y)
+print(results_2)
+# -
+
+x_line = np.array(
+    [
+        biorxiv_published_distances["doc_distances"].min(),
+        biorxiv_published_distances["doc_distances"].max(),
+    ]
+)
+y_line = x_line * results_2.slope + results_2.intercept
+
+# graph here?
+plt.figure(figsize=(6, 5))
+ax = plt.hexbin(
+    biorxiv_published_distances["doc_distances"],
+    biorxiv_published_distances["days_to_published"],
+    gridsize=50,
+    cmap="YlGnBu_r",
+    norm=mpl.colors.LogNorm(),
+    mincnt=1,
+    linewidths=(0.15,)
+    #     edgecolors=None
+)
+ax = plt.gca()
+ax.plot(x_line, y_line, "--k")
+ax.annotate(
+    f"Y={results_2.slope:.2f}*X+{results_2.intercept:.2f}",
+    (8, 1490),
+)
+_ = ax.set_xlabel("Euclidian Distance of Preprint-Published Versions")
+_ = ax.set_ylabel("Time Elapsed Until Preprint is Published (Days)")
+cbar = plt.colorbar()
+_ = cbar.ax.set_ylabel("count", rotation=270)
+_ = ax.scatter(
+    polka_published_preprint_df["doc_distances"],
+    polka_published_preprint_df["days_to_published"],
+    c="red",
+    s=6,
+)
+plt.savefig("output/figures/article_distance_vs_publication_time_hex.svg", dpi=250)
+plt.savefig("output/figures/article_distance_vs_publication_time_hex.png", dpi=250)
```

Comment:
I wonder if the outliers on this plot are the same as the outliers in the SAUCIE plot. Maybe some types of preprint just have a lot of versions (though I guess with n ~ 2 you couldn't really say anything for sure)